lib/util/chat/chat_bubble.dart

```diff
@@ -0,0 +1,240 @@
+import 'package:flutter/material.dart';
+
+class ChatBubble extends StatelessWidget {
+  final bool isSender;
+  final String text;
+  final bool tail;
+  final bool read;
+  final String time;
+
+  const ChatBubble({
+    Key? key,
+    this.isSender = true,
+    required this.text,
+    this.tail = true,
+    this.read = false,
+    required this.time,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    Icon? stateIcon = isSender
+        ? Icon(
+            Icons.done_all,
+            size: 18,
+            color: read ? Theme.of(context).primaryColor : Theme.of(context).colorScheme.onSurface.withOpacity(0.5),
+          )
+        : null;
+
+    return Align(
+      alignment: isSender ? Alignment.topRight : Alignment.topLeft,
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 2),
+        child: CustomPaint(
+          painter: ChatBubblePainter(
+            color: isSender ? Theme.of(context).colorScheme.onSurface.withOpacity(0.1) : Theme.of(context).primaryColor,
+            alignment: isSender ? Alignment.topRight : Alignment.topLeft,
+            tail: tail,
+          ),
+          child: Container(
+            constraints: BoxConstraints(
+              maxWidth: MediaQuery.of(context).size.width * .7,
+            ),
+            margin: isSender ? const EdgeInsets.fromLTRB(7, 7, 14, 7) : const EdgeInsets.fromLTRB(17, 7, 7, 7),
+            child: Stack(
+              children: <Widget>[
+                Padding(
+                  padding: stateIcon != null
+                      ? const EdgeInsets.only(left: 4, right: 20)
+                      : const EdgeInsets.only(left: 4, right: 4),
+                  child: Text(
+                    text,
+                    textAlign: TextAlign.left,
+                  ),
+                ),
+                stateIcon == null
+                    ? const SizedBox(width: 0, height: 0)
+                    : Positioned(
+                        bottom: 0,
+                        right: 0,
+                        child: stateIcon,
+                      )
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class ChatBubblePainter extends CustomPainter {
+  final Color color;
+  final Alignment alignment;
+  final bool tail;
+
+  ChatBubblePainter({
+    required this.color,
+    required this.alignment,
+    required this.tail,
+  });
+
+  final double _radius = 10.0;
+
+  @override
+  void paint(Canvas canvas, Size size) {
```

Comment:
I'm gonna assume you didn't write this code? How easy would it be to change? Because I think the bubbles are slightly too rounded

Reply:
noooo i definitely wrote that code myself. I don't know but you can look [here](https://pub.dev/packages/chat_bubbles) for more designs and strangely also our current design is in that package